projects/components/src/utils/test/cypress-widget-driver.ts

```diff
@@ -0,0 +1,145 @@
+/*!
+ * Copyright 2020 VMware, Inc.
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+import { WidgetDriver } from './base-widget-object';
+import { CypressWidgetFinder } from './cypress-widget-finder';
+import { FindableWidget, FindParams } from './widget-object';
+
+// If we load the 'cypress' module into this file, it also loads Chai which clashes with Jasmine - our testing framework of choice.
+// As such, this class isn't very type safe
+
+/**
+ * A WidgetDriver that knows how to interface with the Cypress framework.
+ */
+export class CypressWidgetDriver implements WidgetDriver {
+    constructor(private element) {}
+
+    find<W extends WidgetDriver>(params: FindParams<FindableWidget<W>> | FindableWidget<W>): W {
+        return new CypressWidgetFinder().find<W, FindableWidget<W>>(params, this.element);
+    }
+
+    exists(cssSelector: string): Promise<boolean> {
+        return new Promise(resolve => this.getElement(cssSelector).then(val => resolve(!!val)));
+    }
+
+    visible(cssSelector?: string): Promise<boolean> {
+        return this.assert('be.visible', cssSelector);
+    }
+
+    hidden(cssSelector?: string): Promise<boolean> {
+        return this.assert('not.be.visible', cssSelector);
+    }
+
+    disabled(cssSelector?: string): Promise<boolean> {
+        return this.assert('be.disabled', cssSelector);
+    }
+
+    classes(cssSelector?: string): Promise<string[]> {
+        return new Promise(resolve =>
+            this.getElement(cssSelector)
+                .invoke('classes')
+                .then(classes => resolve(classes))
+        );
+    }
+
+    click(cssSelector?: string): Promise<void> {
+        return new Promise(resolve =>
+            this.getElement(cssSelector)
+                .click()
+                .then(() => resolve())
+        );
+    }
+
+    check(cssSelector?: string): Promise<void> {
+        return new Promise(resolve =>
+            this.getElement(cssSelector)
+                .check()
+                .then(() => resolve())
+        );
+    }
+
+    getChecked(cssSelector?: string): Promise<boolean> {
+        return this.assert('be.checked', cssSelector);
+    }
+
+    select(option: string, cssSelector?: string): Promise<void> {
```

Comment:
This should be made `async` as I mentioned. But why didn't you just `return this.getElement(cssSelector).select(option)` ? Usually, creating Promise is an anti-pattern. See https://stackoverflow.com/questions/23803743/what-is-the-explicit-promise-construction-antipattern-and-how-do-i-avoid-it